src/test/java/Problem3Test.java

```diff
@@ -7,11 +7,29 @@
     @Test
     public void catchTheBugInBook() {
         // quiz
+        BookFiction ff = new BookFiction("f1", "au1", "g1");
+        BookFiction ff2 = new BookFiction(ff);
+        ff2.setTitle("f2");
+        assertTrue(ff.equals(ff2));
+
+        BookRomance fr = new BookRomance("r1", "au1");
+        BookRomance fr2 = new BookRomance(fr);
+        fr2.setTitle("r2");
+        assertTrue(fr.equals(fr2));
     }
 
     @Test
     public void catchTheBugInMovie() {
         // quiz
+        MovieAction ma = new MovieAction("g", "a1");
+        MovieAction ma2 = new MovieAction(ma);
+        ma2.setTitle("ma2");
+        assertTrue(ma.equals(ma2));
+
+        MovieComedy mc = new MovieComedy("g", "c1");
+        MovieComedy mc2 = new MovieComedy(mc);
+        mc2.setTitle("c2");
+        assertTrue(mc.equals(mc2));
```

Comment:
Well done! I recommend adding checks to make sure there is a false return when the IDs are different